modules/videoio/src/plugin_capture_api.hpp

```diff
@@ -121,6 +121,23 @@ struct OpenCV_VideoIO_Capture_Plugin_API_v1_1_api_entries
         CV_OUT CvPluginCapture* handle);
 }; // OpenCV_VideoIO_Capture_Plugin_API_v1_1_api_entries
 
+struct OpenCV_VideoIO_Capture_Plugin_API_v1_2_api_entries
+{
+    /** @brief Open video capture from buffer with parameters
+
+    @param buffer Memory buffer
+    @param params pointer on 2*n_params array of 'key,value' pairs
+    @param n_params number of passed parameters
+    @param[out] handle pointer on Capture handle
+
+    @note API-CALL 9, API-Version == 2
+     */
+    CvResult (CV_API_CALL *Capture_open_buffer)(
+        std::streambuf& buffer,
```

Comment:
We should not pass C++ objects through low level plugin API. They are not usable in general.

Different C++ ABIs are possible for OpenCV and plugin parts:
- MSVS vs MinGW
- GCC/Clang with different C++ standards
- libc++ (GCC) / libstdc++ (Clang)

Reply:
Thanks! That was my concern. Is that a good idea to replace with functions pointers for required methods like `seek` and `read`?